daft/daft/__init__.pyi

```diff
@@ -1123,29 +1123,20 @@ def interval_lit(
 ) -> PyExpr: ...
 def decimal_lit(sign: bool, digits: tuple[int, ...], exp: int) -> PyExpr: ...
 def series_lit(item: PySeries) -> PyExpr: ...
-def stateless_udf(
+def udf(
     name: str,
-    partial_stateless_udf: PartialStatelessUDF,
+    inner: UninitializedUdf,
+    bound_args: BoundUDFArgs,
     expressions: list[PyExpr],
     return_dtype: PyDataType,
-    resource_request: ResourceRequest | None,
-    batch_size: int | None,
-) -> PyExpr: ...
-def stateful_udf(
-    name: str,
-    partial_stateful_udf: PartialStatefulUDF,
-    expressions: list[PyExpr],
-    return_dtype: PyDataType,
-    resource_request: ResourceRequest | None,
     init_args: InitArgsType,
+    resource_request: ResourceRequest | None,
     batch_size: int | None,
     concurrency: int | None,
 ) -> PyExpr: ...
 def check_column_name_validity(name: str, schema: PySchema): ...
-def extract_partial_stateful_udf_py(
-    expression: PyExpr,
-) -> dict[str, tuple[PartialStatefulUDF, InitArgsType]]: ...
-def bind_stateful_udfs(expression: PyExpr, initialized_funcs: dict[str, Callable]) -> PyExpr: ...
+def initialize_udfs(expression: PyExpr) -> PyExpr: ...
```

Comment:
I think the errors still look pretty reasonable. Here's an example from one of our tests:
<img width="727" alt="image" src="https://github.com/user-attachments/assets/cc21e49f-0c07-40ea-8e8d-340f9433b36a">

Reply:
For actor pool UDFs it's not as nice but that has always been the case. The native executor actually does a decent job but since things are in a separate process it's just not possible to do it as well